src/pkg/services/m365/api/lists.go

```diff
@@ -251,42 +251,49 @@ func (c Lists) PostList(
 		BySiteId(siteID).
 		Lists().
 		Post(ctx, newList, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating list")
+	}
+
+	listItems, err := c.PostListItems(ctx, siteID, restoredList)
```

Comment:
Question: if the restored list already contains the items, were they already included in the restore when you POST the newList?

Reply:
When we post the newList, we do not include the items within it.
We create the `storedList` from `storedListBytes`, but we create new empty list instance and populate only metadata items like columns, contentTypes and other list attributes to the newList.